org.eclipse.jdt.core.manipulation/core extension/org/eclipse/jdt/internal/corext/refactoring/structure/ExtractInterfaceProcessor.java

```diff
@@ -1043,6 +1048,14 @@ protected void rewriteTypeOccurrences(final TextEditBasedChangeManager manager,
 					JavaManipulationPlugin.log(exception);
 				}
 				subUnit.getBuffer().setContents(document.get());
+				IPackageFragment fragment = this.getPackageFragment();
+				if (fragment != null && !fragment.equals(this.fSubType.getPackageFragment())) {
```

Comment:
if the destination package is not the same as the subtype's package, we should manaully add a import statement about the new interface so that bindings can be resolved correctly.